src/components/AnacondaPage.jsx

```diff
@@ -30,6 +32,17 @@ export const AnacondaPage = ({ children, setIsFormDisabled, step, title, usePage
         }
     }, [setIsFormDisabled, usePageInit]);
 
+    useEffect(() => {
+        if (!isFormDisabled && !showPageRef.current) {
+            showPageRef.current = true;
+            setShowPage(true);
+        }
+    }, [isFormDisabled]);
+
+    if (!showPage) {
+        return null;
```

Comment:
Could it happen that initialization takes very long (like the Storage page)? Staring at a blank page isn't very nice either, perhaps this should be rendered to an empty state after 100ms timeout or so?

Reply:
It's not noticeable by the user at this stage, but it can lead into race conditions inside pages, if we dont guard it. Happy to send a follow up to show the empty state.